addons/l10n_sa/data/account_tax_template_data.xml

```diff
@@ -0,0 +1,908 @@
+            <field name="chart_template_id" ref="sa_chart_template_standard"/>
+            <field name="invoice_repartition_line_ids" eval="[(5, 0, 0),
+            (0,0, {
+                'factor_percent': 100,
+                'repartition_type': 'base',
+                'plus_report_line_ids': [ref('tax_report_line_zero_rated_purchases_base')],
+            }),
+            (0,0, {
+                'factor_percent': 100,
+                'repartition_type': 'tax',
+                'plus_report_line_ids': [ref('tax_report_line_zero_rated_purchases_tax')],
+            }),
+        ]"/>
+            <field name="refund_repartition_line_ids" eval="[(5, 0, 0),
+            (0,0, {
+                'factor_percent': 100,
+                'repartition_type': 'base',
+                'minus_report_line_ids': [ref('tax_report_line_zero_rated_purchases_base')],
+            }),
+            (0,0, {
+                'factor_percent': 100,
+                'repartition_type': 'tax',
+                'minus_report_line_ids': [ref('tax_report_line_zero_rated_purchases_tax')],
+            }),
+        ]"/>
+        </record>
+        <record id="sa_exempt_purchases_tax" model="account.tax.template">
+            <field name="name">Exempt Purchases</field>
+            <field name="type_tax_use">purchase</field>
+            <field name="amount">0</field>
+            <field name="amount_type">percent</field>
+            <field name="description">Exempt Purchases</field>
+            <field name="tax_group_id" ref="sa_tax_group_taxes_other"/>
+            <field name="chart_template_id" ref="sa_chart_template_standard"/>
+            <field name="invoice_repartition_line_ids" eval="[(5, 0, 0),
+            (0,0, {
+                'factor_percent': 100,
+                'repartition_type': 'base',
+                'plus_report_line_ids': [ref('tax_report_line_exempt_purchases_base')],
+            }),
+            (0,0, {
+                'factor_percent': 100,
+                'repartition_type': 'tax',
+                'plus_report_line_ids': [ref('tax_report_line_exempt_purchases_tax')],
+            }),
+        ]"/>
+            <field name="refund_repartition_line_ids" eval="[(5, 0, 0),
+            (0,0, {
+                'factor_percent': 100,
+                'repartition_type': 'base',
+                'minus_report_line_ids': [ref('tax_report_line_exempt_purchases_base')],
+            }),
+            (0,0, {
+                'factor_percent': 100,
+                'repartition_type': 'tax',
+                'minus_report_line_ids': [ref('tax_report_line_exempt_purchases_tax')],
+            }),
+        ]"/>
+        </record>
+        <record id="sa_withholding_tax_5_rental" model="account.tax.template">
+            <field name="name">Withholding Tax 5% (Rental)</field>
+            <field name="type_tax_use">purchase</field>
+            <field name="amount">5</field>
```

Comment:
It seems odd to me that a withholding tax is not negative, and only part of its repartition is; are you sure about this setup ? Or is this tax actually regrouping two (the 'normal' tax and the withholding) ?

Also, we introduced in 14.5 a new preceding_subtotal field on tax groups, that is used for withholding taxes in some countries. Not sure it is applicable for Saudi Arabia, but did you consider it ?

Reply:
This has been discussed with @jco-odoo and @nadaseder, Saudi has a special tax requirements where they have to report to two different authorities. They will pay once a year but they have to report the tax for each month also. In this way, we can easily build the report necessary for the company to pay the government